--- emacs-20.4/lisp/mail/mh-utils.el.nmh	Thu Jul 22 17:32:51 1999
+++ emacs-20.4/lisp/mail/mh-utils.el	Thu Jul 22 17:38:01 1999
@@ -606,6 +606,7 @@
 	    (or (mh-path-search exec-path "inc")
 		(mh-path-search '("/usr/local/bin/mh/"
 				  "/usr/local/mh/"
+				  "/usr/bin/" ;Red Hat GNU/Linux
 				  "/usr/bin/mh/" ;Ultrix 4.2
 				  "/usr/new/mh/" ;Ultrix <4.2
 				  "/usr/contrib/mh/bin/" ;BSDI
@@ -617,7 +618,7 @@
       (let ((mh-base mh-progs))
 	(while (let ((dir-name (file-name-nondirectory
 				(directory-file-name mh-base))))
-		 (or (string= "mh" dir-name)
+		 (or (string= "nmh" dir-name)
 		     (string= "bin" dir-name)))
 	  (setq mh-base
 		(file-name-directory (directory-file-name mh-base))))
@@ -629,7 +630,7 @@
 		  ;; components, then look for lib/mh or mh/lib.
 		  (or (mh-path-search
 		       (mapcar (lambda (p) (expand-file-name p mh-base))
-			       '("lib/mh" "etc/nmh" "/etc/nmh" "mh/lib"))
+			       '("lib/nmh" "etc/nmh" "/etc/nmh" "mh/lib"))
 		       "components"
 		       'file-exists-p))))
 	(or (and mh-lib-progs
@@ -637,7 +638,7 @@
 	    (setq mh-lib-progs
 		  (or (mh-path-search
 		       (mapcar (lambda (p) (expand-file-name p mh-base))
-			       '("lib/mh" "libexec/nmh" "lib/nmh" "mh/lib"))
+			       '("lib/nmh" "libexec/nmh" "lib/mh" "mh/lib"))
 		       "mhl")
 		      (mh-path-search '("/usr/local/bin/mh/") "mhl")
 		      (mh-path-search exec-path "mhl") ;unlikely
