--- lisp/startup.el.locale.alias	Fri Feb 11 16:39:31 2000
+++ lisp/startup.el	Fri Feb 11 16:40:28 2000
@@ -332,14 +332,7 @@
   :group 'auto-save)
 
 (defvar locale-translation-file-name
-  (let ((files '("/usr/lib/X11/locale/locale.alias" ; e.g. X11R6.4
-		 "/usr/X11R6/lib/X11/locale/locale.alias" ; e.g. RedHat 4.2
-		 "/usr/openwin/lib/locale/locale.alias" ; e.g. Solaris 2.6
-		 ;;
-		 ;; The following name appears after the X-related names above,
-		 ;; since the X-related names are what X actually uses.
-		 "/usr/share/locale/locale.alias" ; GNU/Linux sans X
-		 )))
+  (let ((files '("/usr/share/locale/locale.alias")))
     (while (and files (not (file-exists-p (car files))))
       (setq files (cdr files)))
     (car files))
