--- emacs-23.3/configure.in~	2011-08-31 22:32:43.687197466 +0200
+++ emacs-23.3/configure.in	2011-08-31 22:38:04.288619611 +0200
@@ -1998,6 +1998,7 @@
 ### Start of font-backend (under X11) section.
 if test "${HAVE_X11}" = "yes"; then
    PKG_CHECK_MODULES(FONTCONFIG, fontconfig >= 2.2.0, HAVE_FC=yes, HAVE_FC=no)
+   PKG_CHECK_MODULES(FREETYPE, freetype2, HAVE_FREETYPE=yes, HAVE_FREETYPE=no)
 
    ## Use -lXft if available, unless `--with-xft=no'.
    HAVE_XFT=maybe
