--- emacs-20.6.orig/src/coding.h	Fri May 19 23:20:52 2000
+++ emacs-20.6/src/coding.h	Fri May 19 22:58:43 2000
@@ -495,6 +495,9 @@
       s2 = c2 + 0x7E;					\
   } while (0)
 
+/* Must declare this, so that we don't lose high 4 bytes on 64-bit machines.  */
+extern Lisp_Object code_convert_string_norecord ();
+
 /* Encode the file name NAME using the specified coding system
    for file names, if any.  */
 #define ENCODE_FILE(name)						   \

