diff -uNr emacs-20.6.orig/src/keyboard.c emacs-20.6/src/keyboard.c
--- emacs-20.6.orig/src/keyboard.c	Thu May  4 11:31:08 2000
+++ emacs-20.6/src/keyboard.c	Thu May  4 11:37:56 2000
@@ -84,9 +84,9 @@
 
 #ifdef HAVE_WINDOW_SYSTEM
 /* Make all keyboard buffers much bigger when using X windows.  */
-#define KBD_BUFFER_SIZE 4096
+#define KBD_BUFFER_SIZE 16384
 #else	/* No X-windows, character input */
-#define KBD_BUFFER_SIZE 256
+#define KBD_BUFFER_SIZE 16384
 #endif	/* No X-windows */
 
 /* Following definition copied from eval.c */
