--- emacs-20.6/src/widget.c.foo	Fri Jun  2 16:13:23 2000
+++ emacs-20.6/src/widget.c	Fri Jun  2 16:12:00 2000
@@ -538,7 +538,7 @@
 		 XtNheightInc, ch,
 		 XtNminWidth, base_width + min_cols * cw,
 		 XtNminHeight, base_height + min_rows * ch,
-		 0);
+		 NULL);
 }
 
 static void
--- emacs-20.6/lwlib/lwlib.c.foo	Fri Jun  2 15:59:56 2000
+++ emacs-20.6/lwlib/lwlib.c	Fri Jun  2 16:00:26 2000
@@ -29,6 +29,7 @@
 #include "lwlib-utils.h"
 #include <X11/StringDefs.h>
 
+#include <stdlib.h>
 #ifdef __osf__
 #include <string.h>
 #include <stdlib.h>
