--- emacs-22.1/src/s/gnu-linux.h.orig	2008-01-03 13:12:45.000000000 +0000
+++ emacs-22.1/src/s/gnu-linux.h	2008-01-03 13:13:20.000000000 +0000
@@ -168,7 +168,7 @@
 /* Ask GCC where to find libgcc.a.  */
 #define LIB_GCC `$(CC) $(C_SWITCH_X_SITE) -print-libgcc-file-name`
 
-#define START_FILES pre-crt0.o /usr/lib/crt1.o /usr/lib/crti.o
+#define START_FILES pre-crt0.o /usr/lib64/crt1.o /usr/lib64/crti.o
 
 /* Here is how to find X Windows.  LD_SWITCH_X_SITE_AUX gives an -R option
    says where to find X windows at run time.  */
@@ -198,7 +198,7 @@
 #define LIBS_DEBUG
 #undef LIB_GCC
 #define LIB_GCC
-#define LIB_STANDARD -lgcc -lc -lgcc /usr/lib/crtn.o
+#define LIB_STANDARD -lgcc -lc -lgcc /usr/lib64/crtn.o
 
 /* Don't use -g in test compiles in configure.
    This is so we will use the same shared libs for that linking

