--- emacs-24.3/configure.ac.orig	2013-03-05 08:24:59.000000000 +0000
+++ emacs-24.3/configure.ac	2013-04-17 07:09:43.617571656 +0000
@@ -2403,6 +2403,7 @@
 ### Start of font-backend (under X11) section.
 if test "${HAVE_X11}" = "yes"; then
    PKG_CHECK_MODULES(FONTCONFIG, fontconfig >= 2.2.0, HAVE_FC=yes, HAVE_FC=no)
+   PKG_CHECK_MODULES(FREETYPE, freetype2, HAVE_FREETYPE=yes, HAVE_FREETYPE=no)
 
    ## Use -lXft if available, unless `--with-xft=no'.
    HAVE_XFT=maybe
