diff -urN emacs-21.1.95.org/international/lisp/mule-conf.el emacs-21.1.95/international/lisp/mule-conf.el
--- emacs-21.1.95.org/international/lisp/mule-conf.el	Tue Feb 26 21:51:44 2002
+++ emacs-21.1.95/international/lisp/mule-conf.el	Tue Feb 26 21:52:16 2002
@@ -417,8 +417,7 @@
  '((ascii t) (latin-iso8859-1 katakana-jisx0201 t) t t
    nil ascii-eol ascii-cntl nil locking-shift single-shift nil nil nil
    init-bol nil nil)
- '((safe-charsets . t)
-   (mime-charset . x-ctext)))
+ '((safe-charsets . t)))
 
 (defvar non-standard-icccm-encodings-alist
   '(("ISO8859-15" . latin-iso8859-15)
