--- emacs-20.5/lisp/loadup.el.loadup	Fri Feb 11 16:22:32 2000
+++ emacs-20.5/lisp/loadup.el	Fri Feb 11 16:24:23 2000
@@ -198,7 +199,9 @@
       (if (memq system-type '(ms-dos windows-nt))
 	  (setq name (expand-file-name
 		      (if (fboundp 'x-create-frame) "DOC-X" "DOC") "../etc"))
-	(setq name (concat (expand-file-name "../etc/DOC-") name))
+	(setq name (concat (expand-file-name
+			    (if (fboundp 'x-create-frame) "../etc/DOC-X-"
+			      "../etc/DOC-")) name))
 	(if (file-exists-p name)
 	    (delete-file name))
 	(copy-file (expand-file-name "../etc/DOC") name t))
